### PR TITLE
add cr2 its post-training recipe

### PR DESCRIPTION
**Pull request by ParisZhang11**
## Description

Brief description of the changes in this PR.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement

## Changes Made

- [ ] Added/updated documentation
- [ ] Added/updated examples
- [ ] Fixed bugs or issues
- [ ] Improved code quality
- [ ] Updated dependencies

## Testing

- [ ] I have tested the changes locally
- [ ] Documentation builds successfully
- [ ] Pre-commit hooks pass
- [ ] Examples run without errors
- [ ] Links and references are valid

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published

## Additional Notes

Any additional information that reviewers should know.

### CODE REVIEW

File: docs/recipes/post_training/reason2/intelligent-transportation/post_training.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+Overall, we see the accuracy jumps over 92% for both experiments after 1 epoch of training. For 3k tokens, accuracy keeps increasing over 3 epochs of training and reaches 93.65% in the end, while for 8k tokens, the model is more prone to overfitting with more training epochs. For this task and dataset, 3k vision tokens give us overall better performance and will converge faster and provide faster inference times.
+
+<img src="assets/quan_result.png" width="600"/>
+
+### Training Time
+
+We ran all the experiments on 1 node (8 GPUs) of A100. The table below captures the training time for the two different settings with one training epoch. As expected, with 3k vision tokens, the model converged roughly 3 times as fast as 8k vision tokens. In summary, you can train a very accurate model using this amount of data in an hour or less.
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/post_training.md]: The table header says:  Training Time for 3 epochs
But the paragraph says: “training time … with one training epoch”

File: docs/recipes/post_training/reason2/intelligent-transportation/post_training.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+### FP8 Quantization
+
+The script to quantize the model to FP8 is provided in the NVIDIA [Cosmos Reason 2 repo](https://github.com/nvidia-cosmos/cosmos-reason2?tab=readme-ov-file#quantization).
+
+1. Clone the Cosmos Reason 2 repo.
+
+2. Run the quantization script.
+
+   ```shell
+   python ./scripts/quantize.py --model_id '/path/to/post-trained/reason2' --save_dir '/path/to/quantized/model --precision fp8'
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/post_training.md]: Missing a separator

Suggested:
```
shell
python ./scripts/quantize.py \
  --model_id "/path/to/post-trained/reason2" \
  --save_dir "/path/to/quantized/model" \
  --precision fp8
```

File: docs/recipes/post_training/reason2/intelligent-transportation/post_training.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+export CUSTOM_WEIGHTS=/path/to/post-trained/model
+docker run -it --rm --name=cosmos-reason2-8b \
+    --gpus all \
+    --shm-size=32GB \
+    -e NIM_MODEL_NAME=$CUSTOM_WEIGHTS \
+    -e NIM_SERVED_MODEL_NAME="cosmos-reason2-8b" \
+    -v $CUSTOM_WEIGHTS:$CUSTOM_WEIGHTS \
+    -u $(id -u) \
+    -p 8000:8000 \
+    $NIM_IMAGE
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/post_training.md:204]: $NIM_IMAGE is not define.

Suggestion: 
In the above paragraph it will be good to add something like "Set $NIM_IMAGE to the Cosmos Reason2 NIM container image from NGC."

File: docs/recipes/post_training/reason2/intelligent-transportation/post_training.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+#### Calculation for "fps 1, total pixels 8,388,608" to 8k vision tokens
+
+1. The video length is about 75 seconds. With 1 fps, the model samples 74 frames (rounded to the nearest even number).
+2. As specified, the video's total pixel is 8,388,608. Considering the temporal step of 2, the frame pixel limit is 8,388,608 // (74 / 2) = 226,719 pixels. Each frame is resized to approximately 608 $\times$ 352.
+3. The total number of vision tokens is 74 frames / 2 $\times$ 608 $\times$ 352 / (32 $\times$ 32) = 7,733 tokens.
+
+### Model Evaluation
+
+After training, we evaluate the model on the validation set of the Environment VQA subset of the WTS dataset. The evaluation script below will save the model responses and accuracy score in the `results` directory.
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/post_training.md]: Is it possible to add a short phrase about how the ```Validation set``` was formed?

File: docs/recipes/post_training/reason2/intelligent-transportation/post_training.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+<br>
+
+## Results
+
+### Quantitative Results
+
+First, let's review the quantitative results on the environment VQA validation set of the WTS dataset for each experiment: one with "nframes 8" (3k vision tokens) and another with "fps 1, total pixels 8,388,608" (8k vision tokens). Similar to the training set, the validation set is also a collection of multiple choice questions (MCQ) on traffic and pedestrian videos, which are unseen during training.
+
+Overall, we see the accuracy jumps over 92% for both experiments after 1 epoch of training. For 3k tokens, accuracy keeps increasing over 3 epochs of training and reaches 93.65% in the end, while for 8k tokens, the model is more prone to overfitting with more training epochs. For this task and dataset, 3k vision tokens give us overall better performance and will converge faster and provide faster inference times.
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/post_training.md]: This is amazing! Could we remark this result maybe highligthing it, could be in this way.

```3k tokens = fewer frames, higher resolution per frame; 8k tokens = more frames, lower resolution per frame.```

This is just a suggestion.

File: docs/recipes/post_training/reason2/intelligent-transportation/post_training.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+```shell
+# From scripts/examples/reason2/intelligent-transportation directory
+cosmos-rl --config sft_config.toml custom_sft.py
+```
+
+### Hyperparameter optimization
+
+For this SFT experiment, we updated some default configurations in `configs/llava_sft.toml`. Here are the key updates that optimize this experiment on 8 A100.
+
+???+ code "Training Configuration"
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/post_training.md:89]: As I was not able to see this in the rendering document, I cannot validate this admonition is working property.

File: docs/recipes/post_training/reason2/intelligent-transportation/post_training.md [hunk 7]
```diff
@@ -0,0 +1,210 @@
+# Intelligent Transportation Post-Training with Cosmos Reason 2
+
+> **Authors:** [Paris Zhang](https://www.linkedin.com/in/pariszhang11/)
+> **Organization:** NVIDIA
+
+| **Model** | **Workload** | **Use Case** |
+|-----------|--------------|--------------|
+| [Cosmos Reason 2](https://github.com/nvidia-cosmos/cosmos-reason2) | Post-training | Intelligent transportation scene understanding |
+
+## Overview
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/post_training.md:10]: It would be really nice to see a quickstar block in the overview, what a developer is expectin to navigate with the recipe:

```
1. Download WTS (WovenTraffic Safety) Dataset
2. Run preprocess
3. Launch SFT (Supervised Fine-Tuning)
4. Evaluate
5. Deploy with NIM```

This is just a suggestion.

File: docs/recipes/post_training/reason2/intelligent-transportation/post_training.md [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+          },
+          {
+              "from": "gpt",
+              "value": "The weather in the video is sunny."
+          }
+      ]
+  }
+  ```
+
+To pre-process the dataset, run the following command:
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/post_training.md]: Please link the data_process.py file.

File: docs/recipes/post_training/reason2/intelligent-transportation/post_training.md [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+
+<br>
+
+## Post-Training with Supervised Fine-Tuning (SFT)
+
+After preprocessing, the WTS dataset is in Llava dataset format and ready for training. To launch training, we follow the cosmos-rl training command in [Cosmos Reason 2 SFT Guide](https://github.com/nvidia-cosmos/cosmos-reason2/blob/main/examples/cosmos_rl/README.md#supervised-fine-tuning-sft):
+
+```shell
+# From scripts/examples/reason2/intelligent-transportation directory
+cosmos-rl --config sft_config.toml custom_sft.py
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/post_training.md:82]: Please add the link of the py file

File: docs/recipes/post_training/reason2/intelligent-transportation/setup.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
+# Setup and System Requirements
+
+This guide covers the setup requirements for post-training Cosmos Reason 2 for intelligent transportation.
+
+## System Requirements
+
+### Minimum Hardware Requirements
+
+- **GPU**: 8 A100, H100, or later GPU (recommended minimum)
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/setup.md]: Suggestion:
**GPU**: 8xA100, H100, or later GPUs (recommended minimum)

File: docs/recipes/post_training/reason2/intelligent-transportation/setup.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+- **Memory**: Sufficient VRAM for model inference
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/setup.md]: Do you have any minimum recommendation? e.g. 80GB? or we can mention it will depend on the batch size.

File: docs/recipes/post_training/reason2/intelligent-transportation/setup.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+- **Storage**: Adequate space for model weights
+
+### Software Requirements
+
+The setup requires the Cosmos Reason 2 and cosmos-rl repositories and Cosmos Reason 2 model to be properly installed and configured.
```
Comment (paularamo) [inline docs/recipes/post_training/reason2/intelligent-transportation/setup.md]: Suggestion: Add the link of Cosmos Reason 2 repo (Setup Section)

File: docs/recipes/post_training/reason2/intelligent-transportation/post_training.md (hunk 1 above)
Comment (ParisZhang11) [reply docs/recipes/post_training/reason2/intelligent-transportation/post_training.md]: fixed. Thank you!

File: docs/recipes/post_training/reason2/intelligent-transportation/post_training.md (hunk 2 above)
Comment (ParisZhang11) [reply docs/recipes/post_training/reason2/intelligent-transportation/post_training.md]: Fixed
